### DISCUSSION TITLE
Add/remove audio track via SDP renegotiate?

### DISCUSSION THREAD

**Discussion by pavlokdbd** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1967] [Q&A]
Hi everyone! Currently kvsWebRTCClientMaster/Viewer sample apps reject any subsequent SDP offer in case peer connection had been already established via first SDP offer in [samples/Common.c](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/f1073439213db4aa4b4b7cc1423609d920dd5489/samples/Common.c#L1425):
`        case SIGNALING_MESSAGE_TYPE_OFFER:`

`            // Check if we already have an ongoing master session with the same peer`

`            CHK_ERR(!peerConnectionFound, STATUS_INVALID_OPERATION, "Peer connection %s is in progress",
                    pReceivedSignalingMessage->signalingMessage.peerClientId);`

 Is there a way to support renegotiate via another SDP offer, e.g. so that audio track added or removed?
I have tried to disable the check above and create/send answer for new offer in similar way that it is done in `handleOffer()`. `respondWithAnswer()` fails in this case.

**Comment by niyatim23** [comment]
Hi @pavlokdbd, renegotiation is not supported yet

**Comment by pavlokdbd** [comment]
How do you recommend to implement audio with later connection when video is already connected? We need this to have a button for audio in the android app to switch audio on during video session. If we create audio track together with video there is a nasty green dot for microphone which informs the user that audio is beibg recorded and being transmitted to the other party, while we don't want this.
